### PR TITLE
Score and feature extraction additions for California Rainfall hackathon

### PR DESCRIPTION
**Pull request by djgagne**
I have added Brier Score scoring options and a gridded feature extractor for processing netCDF files with xarray that contain multiple ensemble members.

### CODE REVIEW

**Comment by kegl** [comment]
Please add the new starting kit to travis to test the workflow and the scores.

**Comment by agramfort** [comment]
why not using the brier_scrore from sklearn?

**Comment by agramfort** [comment]
see http://scikit-learn.org/stable/modules/generated/sklearn.metrics.brier_score_loss.html

**Comment by djgagne** [comment]
I used my own implementation of Brier Score instead of scikit-learn's since it is very simple to implement (essentially just mean squared error for probabilities). I can replace my implementation with the one in scikit-learn if the team prefers that, but there should not be any differences in performance or results.

**Comment by kegl** [comment]
Yes please. Accuracy or error are even simpler, yet we like to rely on sklearn implementation.

**Comment by djgagne** [comment]
Because Travis CI is timing out on downloading the files for the california_rainfall hackathon, I created a separate [california_rainfall_test](https://github.com/djgagne/california_rainfall_test) repo that uses much smaller versions of the Hackathon files. I would like to have it included in the ramp-kits organization so that it can be tested by ramp-workflow.

**Comment by kegl** [comment]
Please just add a top level comment to the workflow so people know what kind of setup it can be used for (beyond this challenge). Something we did in https://github.com/paris-saclay-cds/ramp-workflow/blob/master/rampwf/workflows/el_nino.py. Otherwise LGTM, @agramfort?

**Comment by djgagne** [comment]
I added a time dimension name lookup in grid_feature_extractor.py so that the code would be a little more generalizable in addition to the top level comments. There was a minor bug in my implemented that I addressed.

File: rampwf/score_types/brier_score.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
+from .base import BaseScoreType
+import numpy as np
+from sklearn.metrics import brier_score_loss
+
+class BrierScore(BaseScoreType):
+    is_lower_the_better = True
+    minimum = 0.0
+    maximum = 1.0
+
+    def __init__(self, name='brier_score', precision=3):
```
Comment (agramfort) [inline rampwf/score_types/brier_score.py]: please add proper docstrings for the functions, methods and classes.

Here it should be

"""
Parameters
------------
name : str
     The human readable name for the score.
precision : int
     The floating point precision of ???
"""

we follow numpydoc formatting guidelines.

see Extended Summary section in : https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt
